go/enclave/rpc/GetTransactionReceipt.go

```diff
@@ -114,6 +114,12 @@ func fetchFromCache(ctx context.Context, storage storage.Storage, cacheService *
 		}
 	}
 	r := marshalReceipt(rec.Receipt, logs, rec.From, rec.To)
+
+	// after the receipt was requested by a user remove it from the cache
+	err := cacheService.DelReceipt(ctx, txHash)
```

Comment:
why?

Reply:
because it means that the user doesn't need it any more. Basically, polling will stop when a receipt is returned